### ISSUE TITLE
compile Nginx with Naxsi WAF support

### ISSUE THREAD

**Issue by mahanmp** [mail-in-a-box/mailinabox#1316]
I saw that the installed nginx contains naxsi WAF related files but when i ran `nginx -V` i doesn't show that it is compiled with Naxsi WAF,

Can we compile the nginx with Naxsi support? 

P.S. I am not running any PHP applications on my server aside from the roundcube but as an additional security, I think Naxsi WAF must be included.

**Comment by JoshData** [comment]
No, we're going to continue to install from packages because a) it's easier and b) we get security updates automatically.

**Comment by mveplus** [comment]
I agree with @JoshData JoshData, about maintaining security updates is important and compiling from source will add a lot of burdens.
But I was thinking myself about adding WAF to my next mail/cloud server, and I'm testing now my first MiaB installation and it seems great fit out of the box. An option would be adding official [Nginx repo](http://nginx.org/en/linux_packages.html#stable), I've just added the repo and updated to the latest stable from v1.4.6 to v.1.12.2 and except some warnings that need sorting out in the config works out of the box. 
I have not tried to compile and add ModSecurity 3.0 as a standalone library @mahanmp. I don't know about Naxsi WAF, but certainly, [ModSecurity 3.0](https://github.com/SpiderLabs/ModSecurity/wiki/Reference-Manual#Installation_for_NGINX) can be installed and maintained for one that wants it, without adding too much additional maintaining overhead.

**Comment by JoshData** [comment]
I don't know what WAF is but is this available in the version of nginx shipped with Ubuntu 18.04?

**Comment by mahanmp** [comment]
@JoshData this is an example of a WAF for nginx 

https://github.com/nbs-system/naxsi

**Comment by JoshData** [comment]
Ok....
